nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -256,4 +263,31 @@ String currentRefName() {
   FileIO fileIO() {
     return fileIO;
   }
+
+  private class NessieCatalogTableBuilder extends BaseMetastoreCatalogTableBuilder {
+
+    private NessieCatalogTableBuilder(TableIdentifier identifier, Schema schema) {
+      super(identifier, schema);
+    }
+
+    @Override
+    public TableMetadata newTableMetadata() {
+      return updateLocation(super.newTableMetadata());
+    }
+
+    @Override
+    public TableMetadata replacedTableMetadata(TableOperations ops) {
+      return updateLocation(super.replacedTableMetadata(ops));
+    }
+
+    private TableMetadata updateLocation(TableMetadata tableMetadata) {
+      if (location() != null) {
+        // create the table at the location specified.
+        return tableMetadata;
+      }
+      // Different tables with same table name can exist across references in Nessie.
+      // To avoid sharing same table path between two tables with same name, use table uuid in the table path.
+      return tableMetadata.updateLocation(tableMetadata.location() + "_" + tableMetadata.uuid());
```

Comment:
I'd really prefer a new, random UUID instead of the table's UUID.

Reply:
ok. Used the random UUID instead of table UUID.